include/ygm/detail/comm.ipp

```diff
@@ -528,6 +638,10 @@ inline void comm::flush_send_buffer(int dest) {
   static size_t counter = 0;
   if (m_vec_send_buffers[dest].size() > 0) {
     mpi_isend_request request;
+
+    m_next_message_id += size();
+    request.id = m_next_message_id;
```

Comment:
`m_next_message_id` is being used and manipulated outside of an `if (config.trace)` block. This made sense for tracking when `MPI_Ireceive`s started and stopped, but currently isn't useful and leads to missing IDs in our trace files.

It is better to have the `tracer` handle the IDs for events and have the `comm` request new ones whenever they are needed to avoid increments of `m_next_message_id` scattered around.